src/blosc2/lazyexpr.py

```diff
@@ -731,6 +745,38 @@ def visit_Call(self, node):
     return newexpression, newoperands
 
 
+def convert_to_slice(expression):
+    """
+    Assumes all operands are of the form o...
+    Parameters
+    ----------
+    expression: str
+
+    Returns
+    -------
+    new_expr : str
+    """
+
+    new_expr = ""
+    skip_to_char = 0
+    for i, expr_i in enumerate(expression):
+        if i < skip_to_char:
+            continue
+        if expr_i == "[":
+            k = expression[i:].find("]")  # start checking from after [
+            slice_convert = expression[i : i + k + 1]  # include [ and ]
+            slicer = eval(f"np.s_{slice_convert}")
+            slicer = (slicer,) if isinstance(slicer, slice) else slicer  # standardise to tuple
+            if any(isinstance(el, str) for el in slicer):  # handle fields
+                raise ValueError("Cannot handle fields for slicing lazy expressions.")
+            slicer = str(slicer)
+            new_expr += ".slice(" + slicer + ")"
```

Comment:
Can you comment on why decided to use `.slice()` instead of `.__getitem__()`?  Should be useful for the future you.